### DISCUSSION TITLE
Why is my conda env not recognizing Dagster when deploying in a container?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18083] [Q&A]
hello! i'm having a problem using  `conda` with dagster, during deployment (very similar issue to this thread I found, but no concrete workaround was provided) <https://discuss.dagster.io/t/8427578/hello-has-anyone-had-issues-with-continuous-deployment-i-m-u>. The issue seems to me that in my dockerfile, `source activate myenv` is added to the `.bashrc`, so developing in the container is perfectly fine if `docker exec -it mycontainer bash` is run as the `conda`env is automatically activated and dagster is recognized. However, I presume this becomes an issue during deployment since `docker exec mycontainer dagster` is run, skipping the `conda` env activation, and dagster is not recognized. Does anyone have an idea of how to work around this?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700160771102779?thread_ts=1700160771.102779&cid=C01U954MEER).*

**Comment by prha** [comment]
Can you include your `Dockerfile`?  

What do you have included for your CMD/ENTRYPOINT?

**Reply by abrahme** [reply]
```
FROM continuumio/miniconda3
ADD environment.yml /tmp/environment.yml
RUN conda env create -f /tmp/environment.yml
# Pull the environment name out of the environment.yml
RUN echo "source activate $(head -1 /tmp/environment.yml | cut -d' ' -f2)" > ~/.bashrc
ENV PATH /opt/conda/envs/$(head -1 /tmp/environment.yml | cut -d' ' -f2)/bin:$PATH
 ```
It is a version of this ^, with no entrypoint or cmd. The `environment.yml` file contains installation for dagster
